### PR TITLE
[RFC] tests work

### PR DESCRIPTION
**Pull request by cordoval**
:baby:

### CODE REVIEW

**Comment by cordoval** [comment]
@igorw i am now writing test for the oauth class, would be interesting

File: src/Stack/OAuth.php [hunk 1]
```diff
@@ -23,9 +23,9 @@ public function handle(Request $request, $type = HttpKernelInterface::MASTER_REQ
     {
         $this->container['session'] = $request->getSession();
 
-        $controller = $this->container['router']->match($request);
```
Comment (igorw) [inline src/Stack/OAuth.php:26]: as per the other PR, this change should be reverted / removed.

File: src/Stack/OAuth/AuthController.php [hunk 2]
```diff
@@ -15,7 +15,12 @@ class AuthController
     private $successUrl;
     private $failureUrl;
 
-    public function __construct(TokenStorageInterface $storage, ServiceInterface $oauth, $successUrl, $failureUrl)
+    public function __construct(
```
Comment (igorw) [inline src/Stack/OAuth/AuthController.php]: please revert such style-related changes.

File: tests/Stack/Tests/OAuthTest.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+class OAuthTest extends \PHPUnit_Framework_TestCase
+{
+    public function setUp()
+    {
+
+    }
+
+    /** @test */
+    public function it_automatically_sets_its_container_configuration_and_handles_specific_auth_requests()
```
Comment (igorw) [inline tests/Stack/Tests/OAuthTest.php]: for consistency, we should probably use camelCase here. that is the established convention, and unless we decide to change it everywhere, I'd rather keep it camelCased.

File: composer.json [hunk 4]
```diff
@@ -30,5 +32,8 @@
     },
     "extra": {
         "branch-alias": { "dev-master": "1.0-dev" }
+    },
+    "config": {
+        "bin-dir": "bin/"
```
Comment (CHH) [inline composer.json]: Why set this to `bin/`?

Comment (cordoval) [inline composer.json]: just because, we want to later on run travis with bin/phpunit ?

File: tests/common/TestCase.php [hunk 5]
```diff
@@ -0,0 +1,11 @@
+<?php
+
+namespace common;
```
Comment (CHH) [inline tests/common/TestCase.php:3]: We don't use lowercase namespace identifiers. When I look at our conventions, I would put this and all test classes into the `Stack\Oauth` namespace.

Comment (cordoval) [inline tests/common/TestCase.php:3]: do you realize that igorw did the same in the other stack middlewares?

File: tests/unit/Stack/RouterTest.php [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        $router = new Router($this->container, $this->map);
+        $request = Request::create('/auth1');
+        list($controller, $action) = $router->match($request);
+        $this->assertInstanceOf('Stack\OAuth\AuthController', $controller);
+        $this->assertEquals('actionA', $action);
+    }
+
+    /** @test */
+    public function it_does_not_match_for_non_mapped_path()
```
Comment (CHH) [inline tests/unit/Stack/RouterTest.php]: Please use camelCase here to be consistent with the rest of the code.

File: tests/unit/Stack/RouterTest.php [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        });
+
+        $this->map = [
+            '/auth1' => 'auth_controller:actionA',
+            '/auth2' => 'auth_controller:actionB',
+        ];
+    }
+
+    /** @test */
+    public function it_matches_when_request_pathinfo_path_is_mapped()
```
Comment (CHH) [inline tests/unit/Stack/RouterTest.php]: Please use camelCase here to be consistent with the rest of the code.

File: tests/common/TestCase.php (hunk 5 above)
Comment (CHH) [inline tests/common/TestCase.php:3]: I'm sorry, just saw it.

Please ignore :wink:

File: composer.json (hunk 4 above)
Comment (CHH) [inline composer.json]: Well, Travis runs it also from `./vendor/bin/phpunit` (the default if `bin-dir` is not set)

Comment (cordoval) [inline composer.json]: let's leave it easy for us devs

File: tests/unit/Stack/RouterTest.php [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+namespace Stack;
+
+use OAuth\Common\Storage\TokenStorageInterface;
+use OAuth\Common\Token\TokenInterface;
+use Pimple;
+use Stack\OAuth\AuthController;
+use Stack\Router;
+use Symfony\Component\HttpFoundation\Request;
+
+class RouterTest extends \PHPUnit_Framework_TestCase
```
Comment (igorw) [inline tests/unit/Stack/RouterTest.php:12]: This is not really a Router unit test, but more of a AuthController integration test.

File: tests/integration/OAuthTest.php [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+use Pimple;
+use Stack\CallableHttpKernel;
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\Storage\MockFileSessionStorage;
+use Symfony\Component\HttpKernel\Client;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+class OAuthTest extends TestCase
```
Comment (igorw) [inline tests/integration/OAuthTest.php:15]: work in progress? :)

Comment (cordoval) [inline tests/integration/OAuthTest.php:15]: it was yeah, if i cross need-paths then i may attempt it.

**Pull request closed without merging**